resources/responses.ts

```diff
@@ -0,0 +1,464 @@
+export const triggerFunctions = [
+  'alarmText',
+  'alertText',
+  'condition',
+  'delaySeconds',
+  'disabled',
+  'durationSeconds',
+  'id',
+  'infoText',
+  'preRun',
+  'promise',
+  'response',
+  'run',
+  'sound',
+  'soundVolume',
+  'suppressSeconds',
+  'tts',
+  'outputStrings',
+];
+
+// Trigger fields that can produce text output.
+export const triggerTextOutputFunctions = [
+  'alarmText',
+  'alertText',
+  'infoText',
+  'response',
+  'tts',
+];
+
+// If a trigger has any of these, then it has a visible/audio effect.
+export const triggerOutputFunctions = [
+  ...triggerTextOutputFunctions,
+  'sound',
+];
+
+export const severityMap: { [sev in Severity]: SevText } = {
+  'info': 'infoText',
+  'alert': 'alertText',
+  'alarm': 'alarmText',
+};
+
+const getText = (sev: Severity): SevText => {
+  if (!(sev in severityMap))
+    throw new Error(`Invalid severity: ${sev}.`);
+  return severityMap[sev];
+};
+
+const defaultInfoText = (sev: Severity): SevText => {
+  if (!sev)
+    return 'infoText';
+  return getText(sev);
+};
+
+const defaultAlertText = (sev: Severity): SevText => {
+  if (!sev)
+    return 'alertText';
+  return getText(sev);
+};
+
+const defaultAlarmText = (sev: Severity): SevText => {
+  if (!sev)
+    return 'alarmText';
+  return getText(sev);
+};
+
+const getTarget = (matches: TargetedMatches) => {
+  // Often tankbusters can be casted by the boss on the boss.
+  // Consider this as "not having a target".
+  if (!matches || matches.target === matches.source)
+    return;
+  return matches.target;
+};
+
+const getSource = (matches: TargetedMatches) => {
+  return matches?.source;
+};
+
+// FIXME: make this work for any number of pairs of params
+const combineFuncs = function(text1: SevText, func1: TargetedFunc,
+    text2: SevText, func2: TargetedFunc) {
+  const obj: TargetedResponseOutput = {};
+
+  if (text1 !== text2) {
+    obj[text1] = func1;
+    obj[text2] = func2;
+  } else {
+    obj[text1] = (data: Data, matches: TargetedMatches, output: Output) => {
+      func1(data, matches, output) || func2(data, matches, output);
+    };
+  }
+  return obj;
+};
+
+const isPlayerId = (id?: string) => {
+  return id && id[0] !== '4';
+};
+
+// For responses that unconditionally return static text.
+const staticResponse = (field: SevText, text: LocaleText): StaticResponseFunc => {
+  return (data: unknown, _: unknown, output: Output) => {
+    // cactbot-builtin-response
+    output.responseOutputStrings = {
+      text: text,
+    };
+    return {
+      [field]: (data: unknown, _: unknown, output: Output) => output.text?.(),
+    };
+  };
+};
+
+type SingleSevToResponseFunc = (sev: Severity) => TargetedResponseFunc | StaticResponseFunc;
+type DoubleSevToResponseFunc = (targetSev: Severity, otherSev: Severity) => TargetedResponseFunc;
+type ResponsesMap = {
+  [response: string]: SingleSevToResponseFunc | DoubleSevToResponseFunc;
+};
+
+export const Responses: ResponsesMap = <const>{
```

Comment:
This was mentioned offline, but should we enforce a styling here for `<...>` or `as ...`?

Reply:
Switched to `as const`.  I'll follow up with the eslint rule.